src/agent/debuglet.ts

```diff
@@ -107,6 +113,52 @@ const formatBreakpoints = function(
           .join('\n');
 };
 
+/**
+ * CachedPromise stores a promise. This promise is initially null and can be
+ * assigned to a new promise when refresh is called. This promise will be
+ * resolved by calling member function resolve.
+ */
+export class CachedPromise {
+  private promise: Promise<void>|null = null;
+  private promiseResolve: (() => void)|null = null;
+
+  refresh(): Promise<void> {
+    this.promise = new Promise<void>((resolve) => {
+      this.promiseResolve = resolve;
+    });
+    return this.promise;
+  }
+  get(): Promise<void>|null {
+    return this.promise;
+  }
+  resolve(): void {
+    // Each promise can be resolved only once.
+    if (this.promiseResolve) {
+      this.promiseResolve();
+      this.promiseResolve = null;
+    }
+  }
+  clear(): void {
+    this.promise = null;
+  }
+}
+export interface IsReady {
+  isReady(): Promise<void>;
+}
+
+/**
+ * IsReadyManager is a wrapper class to call debuglet.isReady().
+ */
+class IsReadyImpl implements IsReady {
```

Comment:
I think our style guide would suggest writing this as:
```typescript
class IsReadyImpl implements IsReady {
  constructor(private debuglet: Debuglet) {}
  isReady(): Promise<void> {
    return this.debuglet.isReady();
  }
}
```

Reply:
Done.